pyproject.toml

```diff
@@ -126,22 +132,42 @@ exclude_lines = [
 "@(abc\\.)?abstractmethod",
 ]
 
-[tool.flake8]
-ignore = "E501, W503, E402"
-builtins = "c, get_config"
-exclude = [
-    ".cache",
-    ".github",
-    "docs",
-]
-enable-extensions = "G"
-extend-ignore = [
-    "G001", "G002", "G004", "G200", "G201", "G202",
-    # black adds spaces around ':'
-    "E203",
+[tool.black]
+line-length = 100
+target-version = ["py37"]
+skip-string-normalization = true
+extend-exclude = "enterprise_gateway/.*ipynb"
+
+[tool.ruff]
+target-version = "py37"
+line-length = 100
+select = [
+  "A", "B", "C", "E", "F", "FBT", "I", "N", "Q", "RUF", "S", "T",
+  "UP", "W", "YTT",
 ]
-per-file-ignores = [
-    # B011: Do not call assert False since python -O removes these calls
-    # F841 local variable 'foo' is assigned to but never used
-    "enterprise_gateway/tests/*: B011", "F841",
+ignore = [
+# Q000 Single quotes found but double quotes preferred
+"Q000",
+# FBT001 Boolean positional arg in function definition
+"FBT001", "FBT002", "FBT003",
+# C901 `async_setup_kernel` is too complex (12)
+"C901",
+# C408 Unnecessary `dict` call (rewrite as a literal)
+"C408",
+# N806 Variable `reasonErr` in function should be lowercase
+"N806", "N802", "N803",
+# I252 Relative imports from parent modules are banned
+"I252",
```

Comment:
I'm seeing a warning for this entry when running `make lint`:
```
warning: `I252` has been remapped to `TID252`
```
```suggestion
# TID252 Relative imports from parent modules are banned
"TID252",
```
and is removed following the suggested change.